core/src/main/java/org/infinispan/util/concurrent/CommandAckCollector.java

```diff
@@ -338,27 +340,27 @@ synchronized final void completeExceptionally(Throwable throwable, int topologyI
 
    private class SingleKeyCollector extends Collector<Object> {
       @GuardedBy("this")
-      private final Collection<Address> owners;
+      private final HashSet<Address> owners;
```

Comment:
I was just curious to see if you saw a difference in perf, I'm fine with `HashSet`.